pkg/build/api/validation/validation.go

```diff
@@ -151,50 +161,27 @@ func hasProxy(source *buildapi.GitBuildSource) bool {
 
 func validateSource(input *buildapi.BuildSource) fielderrors.ValidationErrorList {
 	allErrs := fielderrors.ValidationErrorList{}
-	switch input.Type {
-	case buildapi.BuildSourceGit:
-		if input.Git == nil {
-			allErrs = append(allErrs, fielderrors.NewFieldRequired("git"))
-		} else {
-			allErrs = append(allErrs, validateGitSource(input.Git).Prefix("git")...)
-		}
-		if input.Dockerfile != nil {
-			allErrs = append(allErrs, validateDockerfile(*input.Dockerfile)...)
-		}
+
+	if input.Git != nil {
```

Comment:
If `Git` and `Binary` are set simultaneously, do not call `validateGitSource`, as it may add irrelevant errors to `allErrs`, obfuscating a more critical problem. I propose:

```go
if input.Git != nil {
	if input.Binary != nil {
		return append(allErrs, fielderrors.NewFieldInvalid("binary", "", "git and binary sources may not be used together"))
	}
	allErrs = append(allErrs, validateGitSource(input.Git).Prefix("git")...)
}
```